dojo/tools/drheader/parser.py

```diff
@@ -19,14 +19,49 @@ def get_findings(self, filename, test):
             data = json.load(filename)
         except ValueError as err:
             data = {}
-        for item in data:
-            title = "Header : " + item["rule"]
-            message = item["message"]
-            severity = item["severity"].title()
-            find = Finding(title=title,
-                           test=test,
-                           description=message,
-                           severity=severity,
-                           static_finding=False)
-            items.append(find)
-        return items
+        if data != {} and data[0].get("url") is not None:
+            for item in data:
+                url = item["url"]
+                for finding in item["report"]:
+                    title = "Header : " + finding["rule"]
+                    message = finding["message"] + "\nURL : " + url
+                    if finding.get("value") is not None:
+                        message += "\nObserved values: " + finding["value"]
+                    if finding.get("expected") is not None:
+                        message += "\nExpected values: "
+                        for expect in finding["expected"]:
+                            if expect == finding["expected"][-1]:
+                                message += expect
+                            else:
+                                message += expect + "; "
+                    severity = finding["severity"].title()
+                    find = Finding(title=title,
+                                test=test,
+                                url=url,
```

Comment:
I just noticed that you use `Finding.url`. It might be good idea to avoid it. It might be removed in the future:

https://github.com/DefectDojo/django-DefectDojo/blob/3ef065728d85358f254b56375179494861d8c3be/dojo/models.py#L2149-2153